client/modules/summary.js

```diff
@@ -0,0 +1,36 @@
+import {
+  defaultSummary
+} from './defaults';
+
+const REQUEST_SUM = 'budgeting-sample-app/summary/REQUEST_SUM';
+
+export const actions = {
+  requestSum: data => ({
+    type: REQUEST_SUM,
+    data
+  })
+};
+
+/**
+ * Summary calculation
+ * @param  {Object} state  Current state
+ * @param  {Object} action Dispatched action
+ * @return {Object}        Default state
+ */
+export default function reducer(state = defaultSummary, action) {
+  let sum;
+  switch (action.type) {
+    case REQUEST_SUM:
+      sum = action.data.reduce((prev, current) => (
+        { value: prev.value + current.value }
```

Comment:
I don't see the value of using object here. Also I don't see the initial value for reduce. 

How about this:

```js
// some util like math.js 
export const toDecimal = num => (Math.round(parseFloat(num) * 100) / 100).toFixed(2);

// back in this reducer
const balance = toDecimal(action.data.reduce((sum, item) => sum + item.value, 0));
return { ...state, { balance } };
```

One more thing. How about renaming the `reducer` function to something less generic so it's easier to debug?